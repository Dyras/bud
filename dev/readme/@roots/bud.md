## Installation

Install [@roots/bud](/packages/@roots/bud) and [**@roots/bud-cli**](/packages/@roots/bud-cli) to your project

```sh
yarn add @roots/bud @roots/bud-cli --dev
```

## Getting started

Bud can either be configured with a static config file (`json`/`yml`) or a builder module (`js`/`ts`).

Dead simple example:

```js
/**
 * bud.config.js
 */
module.exports = (bud) => bud.entry('app', ['app.js']);
```

Or, as yml:

```yml
# bud.config.yml
entry:
  app: 'app.js'
```

A more advanced configuration might look like:

```ts
/**
 * bud.config.ts
 */
import { Framework } from "@roots/bud";

export default (bud: Framework) =>
  bud
    .use([
      require('@roots/bud-babel'),
      require('@roots/bud-postcss'),
      require('@roots/bud-react'),
    ])
    .setPath('src', 'js')
    .setPublicPath('assets')
    .alias({
      '@src': bud.path('src'),
    })

    .entry({ app: 'app.{js,css}' })

    .when(
      bud.isProduction,
      ({ minimize }) => minimize(),
      ({ devtool }) => devtool()
    );
```

For more on configuring [**@roots/bud**](/packages/@roots/bud) check out the [dedicated documentation](/docs/config/README.md).

## Running a build

![Terminal usage](https://github.com/roots/bud/tree/stable/static/img/cli.svg)

Once you've set up your configuration file the following command will run the build:

```sh
yarn bud build
```

You should see your built assets in the `dist` directory of your project.

### Running in `production` mode

```sh
yarn bud build:production
```

### Running in `development` mode

```sh
yarn bud build:development
```

## Example implementations

There are [example implementations available](/examples).

## Extending

Bud provides an intentionally sparse set of out-of-the-box features.

In fact, much of the core of Bud is actually made up of extensions. This is to make it easy for devs to swap out parts of the framework as needed.

That said, you will likely want to utilize extensions in your project.

If you're unsure where to start or what you need you can try the [**@roots/bud-preset-recommend**](/packages/@roots/bud-preset-recommend) preset. Most require zero configuration.

### First-party extensions

There are a number of Roots maintained extensions available to kickstart your project

| Name                                                                                                                         | Project home                                                                                                                                       | Extension docs                                                                                             | Latest                                                                                                       |
| ---------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
| [**@roots/bud-babel**](/packages/@roots/bud-babel)                                   | [**@babel/babel**](https://github.com/babel/babel)                                                                                                 | [ðŸ“š README](/packages/@roots/bud-babel/README.md)                  | ![npm](https://img.shields.io/npm/v/@roots/bud-babel.svg?color=%23525ddc&style=flat-square)                  |
| [**@roots/bud-compress**](/packages/@roots/bud-compress)                             | [**@webpack-contrib/compression-webpack-plugin**](https://github.com/webpack-contrib/compression-webpack-plugin)                                   | [ðŸ“š README](/packages/@roots/bud-compress/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-compress.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-criticalcss**](/packages/@roots/bud-criticalcss)                       | [**@addyosmani/critical**](https://github.com/addyosmani/critical)                                                                                 | [ðŸ“š README](/packages/@roots/bud-criticalcss/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-criticalcss.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-emotion**](/packages/@roots/bud-emotion)                               | [**@emotion/emotion-css**](https://github.com/emotion/emotion-css)                                                                                 | [ðŸ“š README](/packages/@roots/bud-emotion/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-emotion.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-entrypoints**](/packages/@roots/bud-entrypoints)                       | [**@roots/entrypoints-webpack-plugin**](/packages/@roots/entrypoints-webpack-plugin)                       | [ðŸ“š README](/packages/@roots/bud-entrypoints/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-entrypoints.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-esbuild**](/packages/@roots/bud-esbuild)                               | [**@roots/esbuild-loader**](https://github.com/roots/esbuild-loader)                                                                               | [ðŸ“š README](/packages/@roots/bud-esbuild/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-esbuild.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-eslint**](/packages/@roots/bud-eslint)                                 | [**@webpack-contrib/eslint-webpack-plugin**](https://github.com/webpack-contrib/eslint-webpack-plugin)                                             | [ðŸ“š README](/packages/@roots/bud-eslint/README.md)                 | ![npm](https://img.shields.io/npm/v/@roots/bud-eslint.svg?color=%23525ddc&style=flat-square)                 |
| [**@roots/bud-imagemin**](/packages/@roots/bud-imagemin)                             | [**@webpack-contrib/image-minimizer-webpack-plugin**](https://github.com/webpack-contrib/image-minimizer-webpack-plugin)                           | [ðŸ“š README](/packages/@roots/bud-imagemin/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-imagemin.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-library**](/packages/@roots/bud-library)                               | [**@asfktz/autodll-webpack-plugin**](https://github.com/asfktz/autodll-webpack-plugin)                                                             | [ðŸ“š README](/packages/@roots/bud-library/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-library.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-mdx**](/packages/@roots/bud-mdx)                                       | [**@mdx-js/mdx**](https://github.com/mdx-js/mdx)                                                                                                   | [ðŸ“š README](/packages/@roots/bud-mdx/README.md)                    | ![npm](https://img.shields.io/npm/v/@roots/bud-mdx.svg?color=%23525ddc&style=flat-square)                    |
| [**@roots/bud-postcss**](/packages/@roots/bud-postcss)                               | [**@postcss/postcss**](https://github.com/postcss/postcss)                                                                                         | [ðŸ“š README](/packages/@roots/bud-postcss/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-postcss.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-prettier**](/packages/@roots/bud-prettier)                             | [**@prettier/prettier**](https://github.com/prettier/prettier)                                                                                     | [ðŸ“š README](/packages/@roots/bud-prettier/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-prettier.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-purgecss**](/packages/@roots/bud-purgecss)                             | [**@FullHuman/purgecss**](https://github.com/FullHuman/purgecss)                                                                                   | [ðŸ“š README](/packages/@roots/bud-purgecss/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-purgecss.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-react**](/packages/@roots/bud-react)                                   | [**@facebook/react**](https://github.com/facebook/react)                                                                                           | [ðŸ“š README](/packages/@roots/bud-react/README.md)                  | ![npm](https://img.shields.io/npm/v/@roots/bud-react.svg?color=%23525ddc&style=flat-square)                  |
| [**@roots/bud-sass**](/packages/@roots/bud-sass)                                     | [**@sass/sass**](https://github.com/sass/sass)                                                                                                     | [ðŸ“š README](/packages/@roots/bud-sass/README.md)                   | ![npm](https://img.shields.io/npm/v/@roots/bud-sass.svg?color=%23525ddc&style=flat-square)                   |
| [**@roots/bud-stylelint**](/packages/@roots/bud-stylelint)                           | [**@stylelint/stylelint**](https://github.com/stylelint/stylelint)                                                                                 | [ðŸ“š README](/packages/@roots/bud-stylelint/README.md)              | ![npm](https://img.shields.io/npm/v/@roots/bud-stylelint.svg?color=%23525ddc&style=flat-square)              |
| [**@roots/bud-tailwindcss**](/packages/@roots/bud-tailwindcss)                       | [**@tailwindlabs/tailwindcss**](https://github.com/tailwindlabs/tailwindcss)                                                                       | [ðŸ“š README](/packages/@roots/bud-tailwindcss/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-tailwindcss.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-terser**](/packages/@roots/bud-terser)                                 | [**@terser/terser**](https://github.com/terser/terser)                                                                                             | [ðŸ“š README](/packages/@roots/bud-terser/README.md)                 | ![npm](https://img.shields.io/npm/v/@roots/bud-terser.svg?color=%23525ddc&style=flat-square)                 |
| [**@roots/bud-typescript**](/packages/@roots/bud-typescript)                         | [**@TypeStrong/ts-loader**](https://github.com/TypeStrong/ts-loader)                                                                               | [ðŸ“š README](/packages/@roots/bud-typescript/README.md)             | ![npm](https://img.shields.io/npm/v/@roots/bud-typescript.svg?color=%23525ddc&style=flat-square)             |
| [**@roots/bud-vue**](/packages/@roots/bud-vue)                                       | [**@vue/vue**](https://github.com/vue/vue)                                                                                                         | [ðŸ“š README](/packages/@roots/bud-vue/README.md)                    | ![npm](https://img.shields.io/npm/v/@roots/bud-vue.svg?color=%23525ddc&style=flat-square)                    |
| [**@roots/bud-wordpress-dependencies**](/packages/@roots/bud-wordpress-dependencies) | [**@roots/wordpress-dependencies-webpack-plugin**](/packages/@roots/wordpress-dependencies-webpack-plugin) | [ðŸ“š README](/packages/@roots/bud-wordpress-dependencies/README.md) | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-dependencies.svg?color=%23525ddc&style=flat-square) |
| [**@roots/bud-wordpress-externals**](/packages/@roots/bud-wordpress-externals)       | [**@roots/wordpress-externals-webpack-plugin**](/packages/@roots/wordpress-externals-webpack-plugin)       | [ðŸ“š README](/packages/@roots/bud-wordpress-externals/README.md)    | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-externals.svg?color=%23525ddc&style=flat-square)    |
| [**@roots/bud-wordpress-manifests**](/packages/@roots/bud-wordpress-manifests)       | [**@roots/merged-manifest-webpack-plugin**](/packages/@roots/merged-manifest-webpack-plugin)               | [ðŸ“š README](/packages/@roots/bud-wordpress-manifests/README.md)    | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-manifests.svg?color=%23525ddc&style=flat-square)    |

### Third-party extensions

Have you produced a Bud extension and want to share it here? Please, create an issue sharing information about your project.
