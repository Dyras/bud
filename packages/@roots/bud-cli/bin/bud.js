#!/usr/bin/env node

const CLI = require('../lib/cjs/CLI').CLI
const commands = require('../lib/cjs/CLI/commands')

new CLI({
  command: 'bud',
  projectUrl: 'https://github.com/roots/bud',
  commands,
})
  .mast()
  .invoke()
