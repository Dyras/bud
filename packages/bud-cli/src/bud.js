#!/usr/bin/env node
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bud = void 0;
const bud_support_1 = require("@roots/bud-support");
const build = __importStar(require("./commands/build"));
const publish = __importStar(require("./commands/publish"));
/**
 * $bud
 */
const cli = bud_support_1.yargs
    .command(build)
    .command(publish)
    .recommendCommands()
    .demandCommand(1, 'You must specify a command. See `bud --help` for usage.\n')
    .usage('\nBud-CLI \n\nbud [command] [options]')
    .version()
    .wrap(bud_support_1.yargs.terminalWidth())
    .epilog('https://github.com/roots/bud').argv;
exports.bud = cli;
//# sourceMappingURL=bud.js.map