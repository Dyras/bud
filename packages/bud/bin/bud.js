#!/usr/bin/env node

require('../lib/cjs').bud.cli.invoke()
