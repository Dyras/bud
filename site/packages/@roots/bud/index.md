---
id: "index"
title: "@roots/bud"
slug: "/bud"
sidebar_label: "Exports"
sidebar_position: 0.5
custom_edit_url: null
---

## Namespaces

- [Api](modules/api.md)
- [Build](modules/build.md)
- [Cache](modules/cache.md)
- [CleanWebpackPlugin](modules/cleanwebpackplugin.md)
- [Compiler](modules/compiler.md)
- [CopyWebpackPlugin](modules/copywebpackplugin.md)
- [CssMinimizerWebpackPlugin](modules/cssminimizerwebpackplugin.md)
- [Dashboard](modules/dashboard.md)
- [DefineWebpackPlugin](modules/definewebpackplugin.md)
- [Dependencies](modules/dependencies.md)
- [Discovery](modules/discovery.md)
- [Env](modules/env.md)
- [Extensions](modules/extensions.md)
- [Framework](modules/framework.md)
- [HotModuleReplacementPlugin](modules/hotmodulereplacementplugin.md)
- [IgnoreEmitWebpackPlugin](modules/ignoreemitwebpackplugin.md)
- [MiniCssExtractPlugin](modules/minicssextractplugin.md)
- [WebpackConfigDumpPlugin](modules/webpackconfigdumpplugin.md)
- [WebpackManifestPlugin](modules/webpackmanifestplugin.md)
- [WebpackProvidePlugin](modules/webpackprovideplugin.md)

## Classes

- [Bud](classes/bud.md)
- [Framework](classes/framework.md)
- [Hooks](classes/hooks.md)
- [Logger](classes/logger.md)
- [Server](classes/server.md)

## Variables

### config

• `Const` **config**: `Configuration`

Base config

#### Defined in

packages/@roots/bud/types/config.d.ts:2

___

### extensions

• `Const` **extensions**: `Object`

Base extensions & webpack plugins.

#### Index signature

▪ [x: `string`]: typeof [`WebpackConfigDumpPlugin`](modules/webpackconfigdumpplugin.md) \| typeof [`DefineWebpackPlugin`](modules/definewebpackplugin.md) \| typeof [`HotModuleReplacementPlugin`](modules/hotmodulereplacementplugin.md) \| typeof [`WebpackManifestPlugin`](modules/webpackmanifestplugin.md) \| typeof [`CleanWebpackPlugin`](modules/cleanwebpackplugin.md) \| typeof [`CopyWebpackPlugin`](modules/copywebpackplugin.md) \| typeof [`CssMinimizerWebpackPlugin`](modules/cssminimizerwebpackplugin.md) \| typeof [`IgnoreEmitWebpackPlugin`](modules/ignoreemitwebpackplugin.md) \| typeof [`MiniCssExtractPlugin`](modules/minicssextractplugin.md) \| typeof [`WebpackProvidePlugin`](modules/webpackprovideplugin.md)

#### Type declaration

| Name | Type |
| :------ | :------ |
| `webpack-config-dump-plugin` | typeof [`WebpackConfigDumpPlugin`](modules/webpackconfigdumpplugin.md) |
| `webpack-hot-module-replacement-plugin` | typeof [`HotModuleReplacementPlugin`](modules/hotmodulereplacementplugin.md) |

#### Defined in

packages/@roots/bud/types/extensions/index.d.ts:11

___

### services

• `Const` **services**: `Services`

Base framework services

#### Defined in

packages/@roots/bud/types/services/index.d.ts:5
