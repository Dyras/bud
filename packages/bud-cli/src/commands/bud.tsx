import {yargs} from '@roots/bud-support'

import * as build from './build'
import * as publish from './publish'

/**
 * $bud
 */
const cli = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(
    1,
    'You must specify a command. See `bud --help` for usage.\n',
  )
  .usage('\nBud \n\nbud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .epilog('https://github.com/roots/bud').argv

export {cli as bud}
