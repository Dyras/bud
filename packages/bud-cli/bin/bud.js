#!/usr/bin/env node

require('../lib/cjs/commands/bud')
