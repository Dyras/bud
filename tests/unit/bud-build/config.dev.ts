import {factory, Framework} from '@roots/bud'

describe('bud.build.config', function () {
  let bud: Framework

  beforeAll(() => {
    bud = factory({
      mode: 'development',
    })
  })

  afterAll(done => {
    bud.close(done)
  })

  it('[development] bud.build.config matches snapshot', () => {
    expect(bud.build.config).toMatchSnapshot({
      bail: true,
      cache: {
        type: 'memory',
      },
      context: expect.any(String),
      devtool: false,
      entry: undefined,
      experiments: {
        lazyCompilation: false,
      },
      externals: undefined,
      infrastructureLogging: {
        appendOnly: true,
        level: 'none',
      },
      mode: 'development',
      module: {
        rules: [
          {
            parser: {
              requireEnsure: false,
            },
            test: /\\\\\\.\\[cm\\]\\?\\(jsx\\?\\|tsx\\?\\)\\$/,
          },
          {
            oneOf: [
              {
                exclude: /\\(node_modules\\|bower_components\\)/,
                generator: {
                  filename: 'assets/[hash][ext][query]',
                },
                test: /\\\\\\.\\(png\\|jpe\\?g\\|gif\\)\\$/,
                type: 'asset/resource',
              },
              {
                exclude: /\\(node_modules\\|bower_components\\)/,
                test: /\\\\\\.\\(ttf\\|otf\\|eot\\|woff2\\?\\|ico\\)\\$/,
                use: [
                  {
                    loader: expect.stringContaining(
                      'resolve-url-loader/index.js',
                    ),
                    options: {
                      root: expect.stringContaining('src'),
                      sourceMap: false,
                    },
                  },
                ],
              },
              {
                exclude: /\\(node_modules\\|bower_components\\)/,
                generator: {
                  filename: 'assets/[hash][ext][query]',
                },
                test: /\\\\\\.svg\\$/,
                type: 'asset/resource',
              },
              {
                test: /\\\\\\.\\(html\\?\\)\\$/,
                use: [
                  {
                    loader: expect.stringContaining(
                      'html-loader/dist/cjs.js',
                    ),
                  },
                ],
              },
              {
                test: /\\\\\\.\\(csv\\|tsv\\)\\$/,
                use: [
                  {
                    loader: expect.stringContaining(
                      'csv-loader/index.js',
                    ),
                  },
                ],
              },
              {
                test: /\\\\\\.xml\\$/,
                use: [
                  {
                    loader: expect.stringContaining(
                      'xml-loader/index.js',
                    ),
                  },
                ],
              },
              {
                parser: {
                  parse: expect.any(Function),
                },
                test: /\\\\\\.toml\\$/,
                type: 'json',
              },
              {
                parser: {
                  parse: expect.any(Function),
                },
                test: /\\\\\\.\\(yaml\\|yml\\)\\$/,
                type: 'json',
              },
              {
                parser: {
                  parse: expect.any(Function),
                },
                test: /\\\\\\.json5\\$/,
                type: 'json',
              },
              {
                exclude: /\\(node_modules\\|bower_components\\)/,
                test: /\\\\\\.css\\$/,
                use: [
                  {
                    loader: expect.stringContaining(
                      'style-loader/dist/cjs.js',
                    ),
                  },
                  {
                    loader: expect.stringContaining(
                      'css-loader/dist/cjs.js',
                    ),
                    options: {
                      importLoaders: 1,
                      sourceMap: false,
                    },
                  },
                ],
              },
              {
                exclude: /\\(node_modules\\|bower_components\\)/,
                test: /\\\\\\.\\(js\\|jsx\\)\\$/,
                use: [],
              },
            ],
          },
        ],
      },
      name: 'bud',
      node: false,
      optimization: {
        emitOnErrors: false,
        minimize: false,
        minimizer: ['...'],
        moduleIds: 'deterministic',
        runtimeChunk: undefined,
        splitChunks: {
          cacheGroups: {
            vendor: {
              chunks: 'all',
              filename: 'vendor/[name].js',
              priority: -10,
              reuseExistingChunk: true,
              test: /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]/,
            },
          },
        },
      },
      output: {
        filename: '[name].js',
        path: expect.stringContaining('dist'),
        pathinfo: undefined,
        publicPath: '',
      },
      parallelism: undefined,
      performance: {},
      plugins: [
        {
          apply: expect.any(Function),
          cleanAfterEveryBuildPatterns: [],
          cleanOnceBeforeBuildPatterns: ['**/*', '!dll'],
          cleanStaleWebpackAssets: true,
          currentAssets: [],
          dangerouslyAllowCleanPatternsOutsideProject: false,
          dry: false,
          handleDone: expect.any(Function),
          handleInitial: expect.any(Function),
          initialClean: false,
          outputPath: '',
          protectWebpackAssets: true,
          removeFiles: expect.any(Function),
          verbose: false,
        },
        {
          depth: 8,
          includeFalseValues: false,
          keepCircularReferences: true,
          name: 'bud.webpack.config.js',
          outputPath: expect.stringContaining('.budfiles'),
          showFunctionNames: false,
        },
        {
          options: {},
        },
        {
          DEBUG: false,
          ignorePatterns: [/\\.\\?\\.map\\$/],
          options: {},
        },
        {
          options: {
            basePath: '',
            fileName: 'manifest.json',
            filter: null,
            generate: undefined,
            map: null,
            publicPath: '',
            removeKeyHash:
              /\\(\\[a-f0-9\\]\\{16,32\\}\\\\\\.\\?\\)/gi,
            seed: undefined,
            serialize: expect.any(Function),
            sort: null,
            transformExtensions: /\\^\\(gz\\|map\\)\\$/i,
            useEntryKeys: false,
            useLegacyEmit: false,
            writeToFileEmit: true,
          },
        },
      ],
      profile: false,
      recordsPath: expect.any(String),
      resolve: {
        alias: {},
        extensions: [
          '.wasm',
          '.mjs',
          '.js',
          '.jsx',
          '.css',
          '.json',
          '.json5',
          '.toml',
          '.xml',
          '.csv',
          '.tsv',
          '.yml',
          '.yaml',
          '.xml',
        ],
        modules: ['src', 'node_modules'],
      },
      stats: {},
      target: undefined,
      watch: false,
      watchOptions: undefined,
    })
  })
})
