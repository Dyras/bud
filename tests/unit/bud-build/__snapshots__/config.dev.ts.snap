// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bud.build.config [development] bud.build.config matches snapshot 1`] = `
Object {
  "bail": true,
  "cache": Object {
    "type": "memory",
  },
  "context": Any<String>,
  "devtool": false,
  "entry": undefined,
  "experiments": Object {
    "lazyCompilation": false,
  },
  "externals": undefined,
  "infrastructureLogging": Object {
    "appendOnly": true,
    "level": "none",
  },
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "parser": Object {
          "requireEnsure": false,
        },
        "test": Object {},
      },
      Object {
        "oneOf": Array [
          Object {
            "exclude": Object {},
            "generator": Object {
              "filename": "assets/[hash][ext][query]",
            },
            "test": Object {},
            "type": "asset/resource",
          },
          Object {
            "exclude": Object {},
            "test": Object {},
            "use": Array [
              Object {
                "loader": StringContaining "resolve-url-loader/index.js",
                "options": Object {
                  "root": StringContaining "src",
                  "sourceMap": false,
                },
              },
            ],
          },
          Object {
            "exclude": Object {},
            "generator": Object {
              "filename": "assets/[hash][ext][query]",
            },
            "test": Object {},
            "type": "asset/resource",
          },
          Object {
            "test": Object {},
            "use": Array [
              Object {
                "loader": StringContaining "html-loader/dist/cjs.js",
              },
            ],
          },
          Object {
            "test": Object {},
            "use": Array [
              Object {
                "loader": StringContaining "csv-loader/index.js",
              },
            ],
          },
          Object {
            "test": Object {},
            "use": Array [
              Object {
                "loader": StringContaining "xml-loader/index.js",
              },
            ],
          },
          Object {
            "parser": Object {
              "parse": Any<Function>,
            },
            "test": Object {},
            "type": "json",
          },
          Object {
            "parser": Object {
              "parse": Any<Function>,
            },
            "test": Object {},
            "type": "json",
          },
          Object {
            "parser": Object {
              "parse": Any<Function>,
            },
            "test": Object {},
            "type": "json",
          },
          Object {
            "exclude": Object {},
            "test": Object {},
            "use": Array [
              Object {
                "loader": StringContaining "style-loader/dist/cjs.js",
              },
              Object {
                "loader": StringContaining "css-loader/dist/cjs.js",
                "options": Object {
                  "importLoaders": 1,
                  "sourceMap": false,
                },
              },
            ],
          },
          Object {
            "exclude": Object {},
            "test": Object {},
            "use": Array [],
          },
        ],
      },
    ],
  },
  "name": "bud",
  "node": false,
  "optimization": Object {
    "emitOnErrors": false,
    "minimize": false,
    "minimizer": Array [
      "...",
    ],
    "moduleIds": "deterministic",
    "runtimeChunk": undefined,
    "splitChunks": Object {
      "cacheGroups": Object {
        "vendor": Object {
          "chunks": "all",
          "filename": "vendor/[name].js",
          "priority": -10,
          "reuseExistingChunk": true,
          "test": Object {},
        },
      },
    },
  },
  "output": Object {
    "filename": "[name].js",
    "path": StringContaining "dist",
    "pathinfo": undefined,
    "publicPath": "",
  },
  "parallelism": undefined,
  "performance": Object {},
  "plugins": Array [
    Object {
      "apply": Any<Function>,
      "cleanAfterEveryBuildPatterns": Array [],
      "cleanOnceBeforeBuildPatterns": Array [
        "**/*",
        "!dll",
      ],
      "cleanStaleWebpackAssets": true,
      "currentAssets": Array [],
      "dangerouslyAllowCleanPatternsOutsideProject": false,
      "dry": false,
      "handleDone": Any<Function>,
      "handleInitial": Any<Function>,
      "initialClean": false,
      "outputPath": "",
      "protectWebpackAssets": true,
      "removeFiles": Any<Function>,
      "verbose": false,
    },
    Object {
      "depth": 8,
      "includeFalseValues": false,
      "keepCircularReferences": true,
      "name": "bud.webpack.config.js",
      "outputPath": StringContaining ".budfiles",
      "showFunctionNames": true,
    },
    Object {
      "options": Object {},
    },
    Object {
      "DEBUG": false,
      "ignorePatterns": Array [
        Object {},
      ],
      "options": Object {},
    },
    Object {
      "options": Object {
        "basePath": "",
        "fileName": "manifest.json",
        "filter": null,
        "generate": undefined,
        "map": null,
        "publicPath": "",
        "removeKeyHash": Object {},
        "seed": undefined,
        "serialize": Any<Function>,
        "sort": null,
        "transformExtensions": Object {},
        "useEntryKeys": false,
        "useLegacyEmit": false,
        "writeToFileEmit": true,
      },
    },
  ],
  "profile": false,
  "recordsPath": Any<String>,
  "resolve": Object {
    "alias": Object {},
    "extensions": Array [
      ".wasm",
      ".mjs",
      ".js",
      ".jsx",
      ".css",
      ".json",
      ".json5",
      ".toml",
      ".xml",
      ".csv",
      ".tsv",
      ".yml",
      ".yaml",
      ".xml",
    ],
    "modules": Array [
      "src",
      "node_modules",
    ],
  },
  "stats": Object {},
  "target": undefined,
  "watch": false,
  "watchOptions": undefined,
}
`;
